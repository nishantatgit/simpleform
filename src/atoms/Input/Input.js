import React from 'react';

export default function Input(props) {
  const {
    label,
    type,
    id,
    onChange,
    onBlur,
    name,
    value,
    hasError,
    placeholder,
  } = props;

  const onChangeHandler = e => {
    onChange(e);
  };

  return (
    <>
      <label htmlFor={id}>{label}</label>
      <input
        id={id}
        type={type}
        onChange={onChangeHandler}
        onBlur={onBlur}
        value={value}
        className={hasError ? 'has-error' : null}
        name={name}
        placeholder={placeholder}
        aria-invalid={hasError}
      />
    </>
  );
}
