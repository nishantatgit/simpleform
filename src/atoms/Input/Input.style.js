import { css } from 'styled-components';

export default css`
  background-color: pink;
`;
