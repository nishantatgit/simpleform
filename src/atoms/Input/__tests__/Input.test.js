import React from 'react';
import Input from '../Input';
import { render, shallow } from 'enzyme';

describe('<Input />', () => {
  const props = {
    type: 'text',
    label: 'First Name',
    id: 'first-name',
    name: 'firstName',
  };

  it('should render correctly', () => {
    const inputEl = render(<Input {...props} />);
    expect(inputEl).toMatchSnapshot();
  });

  it('fires onChange event handler once per input change', () => {
    const onChange = jest.fn();
    const addonProps = {
      ...props,
      onChange,
    };
    const inputElWithOnChange = shallow(<Input {...addonProps} />);
    inputElWithOnChange.find('input').simulate('change');
    expect(onChange).toHaveBeenCalledTimes(1);
  });

  it('calls onBlur event Handler once on blur', () => {
    const onBlur = jest.fn();
    const addOnProps = {
      ...props,
      onBlur,
    };
    const inputElWithBlur = shallow(<Input {...addOnProps} />);
    inputElWithBlur.find('input').simulate('blur');
    expect(onBlur).toHaveBeenCalledTimes(1);
  });
});
