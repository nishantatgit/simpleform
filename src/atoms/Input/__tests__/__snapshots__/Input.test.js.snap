// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> should render correctly 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "for": "first-name",
    },
    "children": Array [
      Object {
        "data": "First Name",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "text",
      },
    ],
    "name": "label",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": Object {
      "attribs": Object {
        "id": "first-name",
        "name": "firstName",
        "type": "text",
      },
      "children": Array [],
      "name": "input",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": [Circular],
      "root": Object {
        "attribs": Object {},
        "children": Array [
          [Circular],
          [Circular],
        ],
        "name": "root",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": null,
        "prev": null,
        "type": "root",
        "x-attribsNamespace": Object {},
        "x-attribsPrefix": Object {},
      },
      "type": "tag",
      "x-attribsNamespace": Object {
        "id": undefined,
        "name": undefined,
        "type": undefined,
      },
      "x-attribsPrefix": Object {
        "id": undefined,
        "name": undefined,
        "type": undefined,
      },
    },
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
        Object {
          "attribs": Object {
            "id": "first-name",
            "name": "firstName",
            "type": "text",
          },
          "children": Array [],
          "name": "input",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": null,
          "prev": [Circular],
          "root": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "id": undefined,
            "name": undefined,
            "type": undefined,
          },
          "x-attribsPrefix": Object {
            "id": undefined,
            "name": undefined,
            "type": undefined,
          },
        },
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "for": undefined,
    },
    "x-attribsPrefix": Object {
      "for": undefined,
    },
  },
  "1": Object {
    "attribs": Object {
      "id": "first-name",
      "name": "firstName",
      "type": "text",
    },
    "children": Array [],
    "name": "input",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": Object {
      "attribs": Object {
        "for": "first-name",
      },
      "children": Array [
        Object {
          "data": "First Name",
          "next": null,
          "parent": [Circular],
          "prev": null,
          "type": "text",
        },
      ],
      "name": "label",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": [Circular],
      "parent": null,
      "prev": null,
      "root": Object {
        "attribs": Object {},
        "children": Array [
          [Circular],
          [Circular],
        ],
        "name": "root",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": null,
        "prev": null,
        "type": "root",
        "x-attribsNamespace": Object {},
        "x-attribsPrefix": Object {},
      },
      "type": "tag",
      "x-attribsNamespace": Object {
        "for": undefined,
      },
      "x-attribsPrefix": Object {
        "for": undefined,
      },
    },
    "root": Object {
      "attribs": Object {},
      "children": Array [
        Object {
          "attribs": Object {
            "for": "first-name",
          },
          "children": Array [
            Object {
              "data": "First Name",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "label",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": null,
          "prev": null,
          "root": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "for": undefined,
          },
          "x-attribsPrefix": Object {
            "for": undefined,
          },
        },
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "id": undefined,
      "name": undefined,
      "type": undefined,
    },
    "x-attribsPrefix": Object {
      "id": undefined,
      "name": undefined,
      "type": undefined,
    },
  },
  "_root": [Circular],
  "length": 2,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
